test/Field/golden/01-uniques.golden

```diff
@@ -1,10 +1,11 @@
-let a_1 = a_0;
+ext a_1;
```

Comment:
Ah, this looks like a bug.

Reply:
Hmm, `renameProgram` seems to consume `MonadSupply` for each variable.  For instance
```
ext a;
ext b;
...
```
will start numbering variables from `2`.  I need to check if that is indeed a bug.

Reply:
I am not sure if it is a bug.  The above behaviour is due to the fact that we share `MonadSupply` between `typeProgram` and `renameProgram` in `parseProgramFrom`.  Since we expose `MonadSupply` in `parseProgramFrom`, I presume that it is not a bug, just us being extremely careful.

Here is the explanation:

1. When typechecking program for the first time, we assign a fresh unique for each distinct identifier, meaning that the following program:

```
ext a;
ext b;
let a = a;
```

get typechecked as

```
ext a_0;
ext b_1;
let a_0 = a_0;
```

then we do the renaming step, which matches uniques to scopes:

```
ext a_2;
ext b_3;
let a_4 = a_2;
```

Reply:
Well, not a big deal, I guess. I do need uniques to start from `0` for golden tests, but I do a separate renaming pass in the compiler anyway, so `MonadSupply` is not shared with anything there.